source/common/tracing/zipkin/zipkin_tracer_impl.cc

```diff
@@ -107,20 +108,29 @@ Tracing::SpanPtr Driver::startSpan(const Tracing::Config& config, Http::HeaderMa
 
     new_zipkin_span =
         tracer.startSpan(config, request_headers.Host()->value().c_str(), start_time, context);
-
   } else if (request_headers.XB3TraceId() && request_headers.XB3SpanId()) {
+    uint64_t traceId(0);
+    uint64_t spanId(0);
     uint64_t parentId(0);
+    if (!StringUtil::atoul(request_headers.XB3TraceId()->value().c_str(), traceId, 16)) {
+      throw EnvoyException(fmt::format("invalid hex string for XB3TraceId '{}'",
```

Comment:
IIRC you can't safely throw here. It will crash the server. I think you probably want to return a null span or something like that.